packages/data/src/components/use-select/utils.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * WordPress dependencies
+ */
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+const isEqual = ( valueA, valueB ) => {
```

Comment:
Might be worth a brief JSDoc to explain the purpose we're seeking to accomplish with this function.

Reply:
When I see `isEqual`, I think of a deep comparison. Might be good to specify what kind of equality this checks for.

Reply:
Agreed to both points, I'll try and get this added today sometime.

Reply:
We should just use the default export of the shallow equality package.

```
import isShallowEqual from '@wordpress/is-shallow-equal';
```

I am pretty sure it does the same thing.

Reply:
doh ya, tests pass with just `isShallowEqual`... I'll update and great catch!

Reply:
> We should just use the default export of the shallow equality package.

Implemented in 642d672

Reply:
> We should just use the default export of the shallow equality package.
> 
> ```
> import isShallowEqual from '@wordpress/is-shallow-equal';
> ```
> 
> I am pretty sure it does the same thing.

It's true and I think it's fine to ship with it as-is (especially with tests to guard against regressions), and while technically it's the specified behavior, it's probably not exceedingly obvious that in code here we're intentionally anticipating to handle non-object/array values. It could be worth a clarifying comment to that end.

---

_Aside:_ `isShallowEqual` is documented as requiring the arguments be passed:

https://github.com/WordPress/gutenberg/blob/fcbdd20ee17b298ac236f71dc4102020882216df/packages/is-shallow-equal/index.js#L15-L16

But it's tested to handle others:

https://github.com/WordPress/gutenberg/blob/fcbdd20ee17b298ac236f71dc4102020882216df/packages/is-shallow-equal/test/index.js#L136-L147

It should probably allow any type.